pydantic/_internal/_generate_schema.py

```diff
@@ -1228,6 +1230,11 @@ def _apply_annotations(  # noqa: C901
         # expand annotations before we start processing them so that `__prepare_pydantic_annotations` can consume
         # individual items from GroupedMetadata
         annotations = list(_known_annotated_metadata.expand_grouped_metadata(annotations))
+        non_field_infos, field_infos = [a for a in annotations if not isinstance(a, FieldInfo)], [
+            a for a in annotations if isinstance(a, FieldInfo)
+        ]
+        if field_infos:
+            annotations = [*non_field_infos, FieldInfo.merge_field_infos(*field_infos)]
```

Comment:
This re-ordering does have some possible consequences. In particular, each `FieldInfo` currently hooks into `__get_pydantic_core_schema__` on its own so the order matters. But this re-ordering says "even if everything else runs in the order it is put into Annotated, we do special handling for Field() so that it runs at the end".